### PR TITLE
ebi-ait/dcp-ingest-central#313: Added wrangling status and wrangler filters to `All Projects` page

### PR DESCRIPTION
**Pull request by yusra-haider**
closes dcp-313

relevant PR for core: https://github.com/ebi-ait/ingest-core/pull/75

### CODE REVIEW

**Review by amnonkhen — DISMISSED** [review]

**Review by ke4 — APPROVED** [review]
Looks fine by me 👍  💯

**Review by ke4 — DISMISSED** [review]

**Review by amnonkhen — APPROVED** [review]

File: client/src/app/all-projects/all-projects.component.html [hunk 1]
```diff
@@ -5,13 +5,32 @@ <h1 class="vf-text vf-text-heading--2">All Projects</h1>
...
+  <mat-form-field class="search-bar">
+    <mat-icon matPrefix>search</mat-icon>
+    <input matInput placeholder="Search Projects" [(ngModel)]="searchText" (keydown.enter)="onKeyEnter(searchText)">
+    <button mat-button *ngIf="searchText" matSuffix mat-icon-button aria-label="Clear" (click)="searchText=''; onKeyEnter(searchText)">
+      <mat-icon>close</mat-icon>
+    </button>
+  </mat-form-field>
+
+  <mat-form-field>
+    <mat-select placeholder="Filter by status" (selectionChange)="onFilterByState($event)">
```
Comment (amnonkhen) [inline client/src/app/all-projects/all-projects.component.html:20]: status or state?

File: client/src/app/all-projects/all-projects.component.ts [hunk 2]
```diff
@@ -3,10 +3,12 @@ import {MatPaginator} from '@angular/material/paginator';
 import {Project, ProjectColumn} from '../shared/models/project';
 import {IngestService} from '../shared/services/ingest.service';
 import {map} from 'rxjs/operators';
-import {Criteria} from '../shared/models/criteria';
 import {MetadataDataSource} from '../shared/data-sources/metadata-data-source';
 import {PagedData} from '../shared/models/page';
 import {MetadataDocument} from '../shared/models/metadata-document';
+import * as ingestSchema from '../project-form/project-ingest-schema.json';
```
Comment (amnonkhen) [inline client/src/app/all-projects/all-projects.component.ts:9]: I would organise the imports alphabetically as well as group by library imports and our own imports. the IDE can do it for you.

File: client/src/app/all-projects/all-projects.component.ts [hunk 3]
```diff
@@ -69,62 +74,46 @@ export class AllProjectsComponent implements OnInit, OnDestroy {
...
-      operator: 'IS',
-      value: false
-    } as Criteria;
-    params['operator'] = 'and';
-    return this.queryProjects([criteria], params);
+  private getFilteredProjects(params) {
+    return this.ingestService.getFilteredProjects(params).pipe(map(
+      data => {
+        const pagedData: PagedData<MetadataDocument> = {data: [], page: undefined};
+        pagedData.data = data._embedded ? data._embedded.projects : [];
```
Comment (amnonkhen) [inline client/src/app/all-projects/all-projects.component.ts:88]: You could use the elvis operator here.
```suggestion
        pagedData.data = data._embedded?.projects || [];
```

**Review by amnonkhen — CHANGES_REQUESTED** [review]

File: client/src/app/all-projects/all-projects.component.html (hunk 1 above)
Comment (yusra-haider) [reply client/src/app/all-projects/all-projects.component.html:20]: as per the UI designs, it's `status`

**Review by amnonkhen — APPROVED** [review]

**Review by aaclan-ebi — APPROVED** [review]

**Pull request merged**